core/grants.go

```diff
@@ -0,0 +1,194 @@
+package core
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+
+	chainjson "chain/encoding/json"
+	"chain/errors"
+	"chain/net/http/authz"
+	"chain/net/http/httpjson"
+)
+
+// an api-friendly representation of a grant
+type apiGrant struct {
+	GuardType string        `json:"guard_type"`
+	GuardData chainjson.Map `json:"guard_data"`
```

Comment:
In order to get consistent serialization (e.g. sorting map keys) we have to completely deserialize the map here and then reserialize it again. Our chainjson.Map type is really just json.RawMessage with an extra check that the json is a map, so it preserves the original serialization which is not what we want. This line needs to be

```go
GuardData map[string]interface{} `json:"guard_data"`
```

or

```go
GuardData interface{} `json:"guard_data"`
```